### ISSUE TITLE
What even is a copula

### ISSUE THREAD

**Issue by aryamanarora** [UniversalDependencies/UD_Punjabi-PunTB#3]
Many issues coming up following:
- #1
- #2

Underlyingly, the problem is that it's not clear what a copula is per UD guidelines, and it's not clear when the dependency structure should be `nsubj` from predicate or from the verb.

UD guidelines mention 6 different types of copular constructions:
1. Equation (aka identification):  “she is my mother” ਉਹ ਮੇਰੀ ਮਾਂ ਹੈ
2. Attribution: “she is nice” ਉਹ ਚੰਗੀ ਹੈ
3. Location: “she is in the bathroom” ਉਹ ਬਾਥਰੂਮ ਵਿੱਚ ਹੈ
4. Possession: “the book is hers” ਕਿਤਾਬ ਉਸਦੀ ਹੈ
5. Benefaction: “the book is for her” ਕਿਤਾਬ ਉਸ ਲਈ ਹੈ
6. Existence: “there is food (in the kitchen)” (ਰਸੋਈ ਵਿੱਚ) ਖਾਣਾ ਹੈ

All 6 use the same verb in Punjabi.

**Comment by nschneid** [comment]
Note that in English existential constructions the verb is not considered a copula (UniversalDependencies/docs#706); though the lemma is the same I guess it is considered syntactically different per the [third bullet point](https://universaldependencies.org/u/overview/simple-syntax.html#nonverbal-clauses).

**Comment by aryamanarora** [comment]
Right, from what I understand even in languages like Czech (which have no `expl` dependent in existential construction, and use the same verb for all 6, just like Punjabi) the verb is still the head in existential constructions, as given in an example on that page. So I'm not sure what that third bullet point's goal is, because it seems for all languages UDv2 wants existential constructions to have the verb as the head, not the subject.

**Comment by nschneid** [comment]
@dan-zeman Are there languages for which the verb in the existential construction IS treated as a copula?

**Comment by aryamanarora** [comment]
- UniversalDependencies/UD_Punjabi-PunTB#4

**Comment by amir-zeldes** [comment]
Just a note to point out, in English we have different trees for copular:

1. A dog is in the yard (root: yard)

But existential

2. There is a dog in the yard (root: is)

In many languages, the two are much more similar, for example the easiest Hebrew equivalent of both of these has the same words, but word order can make it sound more like 1 or 2:

1. yesh kelev b-a-xatser (exists dog in the yard)
2. yesh b-a-xatser kelev (exists in the yard dog)

There may be other paraphrases of course, but I don't feel any syntactic dependency difference between these. Current UD Hebrew views both as existential, with root "yesh", but if there is no verb at all (zero copula), we always have to make the location be the predicate:

3. ha-kelev b-a-xatser "the dog is in the yard" (root: yard, lit. "the-dog in-the-yard")

**Comment by dan-zeman** [comment]
I suggest to move this issue to the [main issue tracker at the docs repository](https://github.com/UniversalDependencies/docs/issues) since it is about annotation guidelines rather than about bugs in a treebank.

**Comment by dan-zeman** [comment]
> @dan-zeman Are there languages for which the verb in the existential construction IS treated as a copula?

I am not aware of such examples in the current data (which does not necessarily mean they don't exist).

I now lean towards saying that the Czech existential sentences should be treated as if _být_ “to be” were a copula there. But I have not implemented it in the data, nor modified the examples in the documentation. With a location, the situation is similar to what @amir-zeldes describes for Hebrew. Pure existentials without a location will either use the verb _existovat_ “to exist”, which will be treated as a regular intransitive verb, or they will use _být_ “to be”, which can still be tagged `AUX` but it will have to be promoted to the head position (as if the location predicate is elided), so there will be no `cop` relation.
